zipline/data/resample.py

```diff
@@ -45,11 +49,10 @@ def minute_to_session(minute_frame, calendar):
         A DataFrame with the columns `open`, `high`, `low`, `close`, `volume`,
         and `day` (datetime-like).
     """
-    # Group minutes into their respective days. Note that this will
-    # create groups for all trading days in the desired range,
-    # including days with no minute data.
-    return minute_frame.resample(calendar.day,
-                                 how=_MINUTE_TO_SESSION_OHCLV_HOW)
+    how = OrderedDict((c, _MINUTE_TO_SESSION_OHCLV_HOW[c])
+                      for c in minute_frame.columns)
+    return minute_frame.dropna().groupby(calendar.minute_to_session_label).agg(
```

Comment:
this will likely benefit from @nathanwolfe's speedup here: https://github.com/quantopian/zipline/pull/1407